src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -60,8 +60,7 @@ internal class SourceLocalSymbol : LocalSymbol
             this.binder = binder;
             this._containingSymbol = containingSymbol;
             this._identifierToken = identifierToken;
-            this._refKind = refKind;
-            this._typeSyntax = typeSyntax;
+            this._typeSyntax = allowRefKind ? typeSyntax.SkipRef(out this._refKind) : typeSyntax;
```

Comment:
This conditional serves to fix the ref kind to none when ref is not allowed. If ref is not allowed, binding the type (which includes "ref") later will give an appropriate diagnostic.

---
In reply to: [73060261](https://github.com/dotnet/roslyn/pull/12841#discussion_r73060261) [](ancestors = 73060261)